### PR TITLE
[v1.4][W09-B4] Flashy

### PR DESCRIPTION
**Pull request by jethrokuan**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -10,9 +10,9 @@ ifdef::env-github[]
 :tip-caption: :bulb:
 :note-caption: :information_source:
 endif::[]
-:repoURL: https://github.com/se-edu/addressbook-level4/tree/master
+:repoURL: https://github.com/CS2103JAN2018-W09-B4/addressbook-level4/tree/master
 
-By: `Team SE-EDU`      Since: `Jun 2016`      Licence: `MIT`
+By: `W09-B4`      Since: `Jun 2016`      Licence: `MIT`
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: The heading in the DG should be updated as well.

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -781,14 +781,11 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
 == Product Scope
 
 *Target user profile*:
+* has the need for flashcards
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: This section isn't formatted properly when rendered on your GitHub pages.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -781,14 +781,11 @@ See this https://github.com/se-edu/addressbook-level4/pull/599[PR] for the step-
...
-* prefer desktop apps over other types
-* can type fast
-* prefers typing over mouse input
-* is reasonably comfortable using CLI apps
-
-*Value proposition*: manage contacts faster than a typical mouse/GUI driven app
+*Value proposition*: helps user memorise facts better.
 
 [appendix]
 == User Stories
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: Please include the user stories and use cases that you showed me in week 7.

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -60,6 +60,15 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
...
+=== Changing the theme : `theme`
+
+Format: `theme t/THEME`
+
+Examples:
+
+* `theme t/light`
+* `theme t/dark`
+
 === Adding a person: `add`
```
Comment (goweiwen) [inline docs/UserGuide.adoc]: Is this feature still in? Remember to update your User Guide for every change to reduce the chances of any stale information misleading users.

**Review by goweiwen — COMMENTED** [review]
Please leave this PR open (do not close it) as we will use it to review your change until v1.5.

**Important**: As mentioned in the v1.0 review, please add your user-stories and use cases to the developer guide by the next tutorial so I can review them. Until then, I will consider this week's team LO as incomplete.

Keep in mind that since you are not merging your minor changes from v1.0, you will need to add new changes and let me know which feature you wish to consider your minor enhancement.

Please look through the user guide and developer guide together and update the relevant sections. For this week, you are required to think through the product's end behaviour and write the entire UG, for **all** the commands that you want to implement.

File: src/main/java/seedu/address/logic/commands/ExitCommand.java [hunk 5]
```diff
@@ -8,7 +8,8 @@
  */
 public class ExitCommand extends Command {
 
-    public static final String COMMAND_WORD = "exit";
+    public static final String COMMAND_WORD_EXIT = "exit";
+    public static final String COMMAND_WORD_QUIT = "quit";
```
Comment (goweiwen) [inline src/main/java/seedu/address/logic/commands/ExitCommand.java]: You should not name your constants to match their content, as this makes them quite redundant and is not resistant to change. Instead, do something like `COMMAND_WORD` and `COMMAND_WORD_ALIAS`.

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -60,6 +60,15 @@ e.g. typing *`help`* and pressing kbd:[Enter] will open the help window.
 
 Format: `help`
 
+=== Changing the theme : `theme`
```
Comment (goweiwen) [inline docs/UserGuide.adoc]: Good job implementing this enhancement 👍.

File: docs/UserGuide.adoc [hunk 7]
```diff
@@ -73,11 +82,24 @@ Examples:
 * `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`
 * `add n/Betsy Crowe t/friend e/betsycrowe@example.com a/Newgate Prison p/1234567 t/criminal`
 
+=== Adding a flashcard: `addc`
```
Comment (goweiwen) [inline docs/UserGuide.adoc]: Well done for this feature as well 👍.

**Review by goweiwen — COMMENTED** [review]
A few comments on the minor enhancements. Your code base is quite well tested, keep up the good work 👍.

File: docs/DeveloperGuide.adoc [hunk 8]
```diff
@@ -358,6 +358,16 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two * different things.
 // end::undoredo[]
 
+===== Aspect: Support for multiple themes
+* **Alternative 1**: Do not support multiple themes
+** Pros: Consistent user experience with a single theme.
+** Cons: No ability to customize the look and feel of the application.
+* **Alternative 2**: Support multiple themes.
+** Pros: Users get a certain extent of customizability in the application.
+** Cons: More development effort needed to support multiple themes and ensure that they can be loaded.
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: This sounds like simply considering implementing the feature or not. Instead, it should be a consideration faced during implementation. For example, customizable vs preset, or fully themeable vs just changing colours.

File: docs/DeveloperGuide.adoc [hunk 9]
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
...
+* has the need for flashcards
+* has a significant amount of material to memorise
+* have vast amount of different subjects to schedule
 
-By creating this command, you will get a chance to learn how to implement a feature end-to-end, touching all major components of the app.
+*Value proposition*: helps user memorise facts better.
 
-*Scenario:* You are a software maintainer for `addressbook`, as the former developer team has moved on to new projects. The current users of your application have a list of new feature requests that they hope the software will eventually have. The most popular request is to allow adding additional comments/notes about a particular contact, by providing a flexible `remark` field for each contact, rather than relying on tags alone. After designing the specification for the `remark` command, you are convinced that this feature is worth implementing. Your job is to implement the `remark` command.
+[appendix]
+== Product Scope
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc:1030]: Duplicate section?

File: docs/DeveloperGuide.adoc [hunk 10]
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
...
-Edits the remark for the first person to `Likes to drink coffee.`
-* `remark 1 r/` +
-Removes the remark for the first person.
+[appendix]
+== Feature Contribution
 
-==== Step-by-step Instructions
+=== Jethro Kuan
+Major enhancement: Implementing a graph structure to link Cards and Tags so that users are able to tag different cards
+in order to search them efficiently.
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc:1044]: How about putting the end goal in front? The implementation is just a minor detail.

File: docs/DeveloperGuide.adoc [hunk 11]
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
...
-===== [Step 1] Logic: Teach the app to accept 'remark' which does nothing
-Let's start by teaching the application how to parse a `remark` command. We will add the logic of `remark` later.
+Minor enhancement: Implementing a list card command so that users would be able to list the cards available
+for reference
 
-**Main:**
+=== Foo Yong Jie
+Major enhancement: Improving the UI / UX of the application, including implementing separate panels for tags, cards,
+and the display of cards.
+This allows users to use the application efficiently.
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: This major enhancement is a bit vague. Maybe go into detail on what are the main changes you aim to make?

File: docs/DeveloperGuide.adoc [hunk 12]
```diff
@@ -493,340 +503,226 @@ Here are the steps to create a new release.
...
+Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (unlikely to have) - `*`
 
-. Modify `RemarkCommandTest` to test the `RemarkCommand#equals()` method.
-. Add `RemarkCommandParserTest` that tests different boundary values
-for `RemarkCommandParser`.
-. Modify link:{repoURL}/src/test/java/seedu/address/logic/parser/AddressBookParserTest.java[`AddressBookParserTest`] to test that the correct command is generated according to the user input.
+[width="100%",cols="22%,<23%,<25%,<30%",options="header",]
+|=======================================================================
+|Priority |As a ... |I want to ... |So that I can...
+| `* *` | University Student | Create a new card                                        | I can review new material
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: No "must have" user stories? I feel like adding and reviewing cards at least are quite important in a flash card application...

File: docs/DeveloperGuide.adoc [hunk 13]
```diff
@@ -358,6 +358,16 @@ image::UndoRedoActivityDiagram.png[width="650"]
 ** Cons: Requires dealing with commands that have already been undone: We must remember to skip these commands. Violates Single Responsibility Principle and Separation of Concerns as `HistoryManager` now needs to do two * different things.
 // end::undoredo[]
 
+===== Aspect: Support for multiple themes
```
Comment (goweiwen) [inline docs/DeveloperGuide.adoc]: This is a subsection of the Undo/Redo feature. It should be part of a new section describing the implementation of multiple themes.

**Review by goweiwen — COMMENTED** [review]
Good job! Your team looks on track to finishing your planned enhancements.

Make sure your user guide is up to date and ready before user testing happens.

Consider adding some documentation to your developer guide as well (specifically the Implementation section).
